bothan-api/server/src/api/server.rs

```diff
@@ -149,13 +149,18 @@ impl<S: Store> BothanService for BothanServer<S> {
     ) -> Result<Response<GetPricesResponse>, Status> {
         info!("received get price request");
         debug!("request: {:?}", request);
+        let metrics = self.manager.get_metrics();
+        metrics.increment_get_prices_total_requests();
+        let start_time = chrono::Utc::now().timestamp_millis();
         let price_request = request.into_inner();
         let (uuid, price_states) = self
             .manager
             .get_prices(price_request.signal_ids.clone())
             .await
             .map_err(|e| {
                 error!("failed to get prices: {}", e);
+                let elapsed_time = (chrono::Utc::now().timestamp_millis() - start_time) as u64;
```

Comment:
either add an explicit comment here or use TryFrom rather than casting to prevent unwanted wrapping